examples/02_decoding/plot_haxby_grid_search.py

```diff
@@ -37,98 +37,99 @@
+# based on F-test.
+from nilearn.decoding import Decoder
+# Here screening_percentile is set to 1.25 percent, meaning around 500
+# features will be selected with ANOVA.
+decoder = Decoder(estimator='svc', mask=mask_filename, smoothing_fwhm=4,
+                  standardize=True, screening_percentile=1.25)
 
 ###########################################################################
-# Compute prediction scores using cross-validation
+# Fit the Decoder and predict the reponses
 # -------------------------------------------------
-anova_svc.fit(X, y)
-y_pred = anova_svc.predict(X)
-
-from sklearn.model_selection import cross_val_score
+decoder.fit(fmri_niimgs, y)
+y_pred = decoder.predict(fmri_niimgs)
 
-k_range = [10, 15, 30, 50, 150, 300, 500, 1000, 1500, 3000, 5000]
+###########################################################################
+# Compute prediction scores with different values of screening percentile
+# -----------------------------------------------------------------------
+import numpy as np
+sp_range = [1.25, 2.5, 3.75, 7.5, 12.5, 25.0]
 cv_scores = []
-scores_validation = []
-
-for k in k_range:
-    feature_selection.k = k
-    cv_scores.append(np.mean(
-        cross_val_score(anova_svc, X[session < 10], y[session < 10], cv=3)))
-    print("CV score: %.4f" % cv_scores[-1])
-
-    anova_svc.fit(X[session < 10], y[session < 10])
-    y_pred = anova_svc.predict(X[session == 10])
-    scores_validation.append(np.mean(y_pred == y[session == 10]))
-    print("score validation: %.4f" % scores_validation[-1])
+val_scores = []
+
+for sp in sp_range:
+    decoder = Decoder(estimator='svc', mask=mask_filename,
+                      smoothing_fwhm=4, cv=3, standardize=True,
+                      screening_percentile=sp)
+    decoder.fit(index_img(fmri_niimgs, session < 10),
+                y[session < 10])
+    cv_scores.append(np.mean(decoder.cv_scores_['bottle']))
+    print("Sreening Percentile: %.3f" % sp)
+    print("Mean CV score: %.4f" % cv_scores[-1])
+
+    y_pred = decoder.predict(index_img(fmri_niimgs, session == 10))
+    val_scores.append(np.mean(y_pred == y[session == 10]))
+    print("Validation score: %.4f" % val_scores[-1])
 
 ###########################################################################
 # Nested cross-validation
-# -------------------------
-from sklearn.model_selection import GridSearchCV
-# We are going to tune the parameter 'k' of the step called 'anova' in
-# the pipeline. Thus we need to address it as 'anova__k'.
-
-# Note that GridSearchCV takes an n_jobs argument that can make it go
-# much faster
-grid = GridSearchCV(anova_svc, param_grid={'anova__k': k_range}, verbose=1,
-                    cv=3)
-nested_cv_scores = cross_val_score(grid, X, y, cv=3)
+# -----------------------
+# We are going to tune the parameter 'screening_percentile' in the
+# pipeline.
+from sklearn.model_selection import KFold
+cv = KFold(n_splits=3)
+nested_cv_scores = []
+
+for train, test in cv.split(session):
```

Comment:
I would love to have reviews & possible suggestions on this part as the doing the nested CV loop like this is not the most efficient. Scikit-learn `GridSearchCV` and `cross_val_score` cannot be used here since `decoder only take input in the form of Nifty Images file-path/format.

Reply:
You mean, we need change the API to supported nested CV ?

I think I'd rather keep it as is for that release and open an issue to revisit that later.

Reply:
I think binh thinks reimplementing a cross-val here is heavy and I agree. I proposed a compromise below

Reply:
Actually @tbng I just noticed the section above. Maybe it's enough for an example, keeping the plotting section that come last . So I suggest removing the "Nested cross val" section.

Reply:
So, we agree to keep this as-is and leave enhancements for a future PR ?

Reply:
+1 for leaving it to future ```Decoder``` object enhancements.